test/tasks/karpenter-test-suite.yaml

```diff
@@ -0,0 +1,127 @@
+# ginkgo-suite is a generic task that executes ginkgo on the given repository and ginkgo suite.
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: karpenter-test-suite
+  namespace: karpenter-tests
+spec:
+  description: |
+    Execute a karpenter test suite
+  params:
+    - name: cluster-name
+      default: $(context.taskRun.name)
+    - name: repository
+      default: github.com/aws/karpenter
+    - name: tag
+      default: v0.13.2
+    - name: command
+      default: go test ./pkg/...
+  results:
+    - name: account-id
+    - name: region
+    - name: kubeconfig
+  workspaces:
+    - name: ws
+  steps:
+
+  - name: detect-environment
+    image: alpine/curl:3.14
+    workingDir: $(workspaces.ws.path)/$(params.repository)
+    script: |
+      TOKEN=$(curl -X PUT "http://169.254.169.254/latest/api/token" -H "X-aws-ec2-metadata-token-ttl-seconds: 21600")
+      curl -H "X-aws-ec2-metadata-token: $TOKEN" -v http://169.254.169.254/latest/meta-data/identity-credentials/ec2/info | grep "AccountId" | cut -f4 -d"\"" > $(results.account-id.path)
+      curl -H "X-aws-ec2-metadata-token: $TOKEN" -v http://169.254.169.254/latest/meta-data/placement/region > $(results.region.path)
+
+  - name: checkout-repository
+    image: alpine/git
+    workingDir: $(workspaces.ws.path)/$(params.repository)
+    script: |
+      git clone https://$(params.repository).git .
+      git checkout $(params.tag)
+
+  - name: create-iam-policies
+    image: amazon/aws-cli:2.7.11
+    workingDir: $(workspaces.ws.path)/$(params.repository)
+    script: |
+      aws cloudformation deploy \
+        --stack-name "$(params.cluster-name)" \
+        --template-file website/content/en/preview/getting-started/getting-started-with-eksctl/cloudformation.yaml \
+        --capabilities CAPABILITY_NAMED_IAM \
+        --parameter-overrides "ClusterName=$(params.cluster-name)"
+
+  - name: create-cluster-configuration
+    image: alpine/curl:3.14
+    workingDir: $(workspaces.ws.path)/$(params.repository)
+    script: |
+      cat <<EOF > cluster.yaml
```

Comment:
Any problems with putting this in the same step as create-cluster? The only time we should be getting an error here is with the image.

Reply:
Good idea. I'm no longer curling in here.